qemu/tests/qemu_guest_agent.py

```diff
@@ -4168,15 +4195,54 @@ def _base_on_bg_check_resource_leak():
-                              " %s, memory is %s" % (qga_handles, qga_memory))
-                else:
-                    LOG_JOB.info("Current qga handles is %s, qga memory"
-                                 "is %s", qga_handles, qga_memory)
+                if qga_memory > qga_mem_threshold:
+                    qga_memory_previous = qga_memory
+                    qga_handles_previous = qga_handles
+                    qga_handles_latest, qga_memory_latest = _process_qga_resource()
+                    memory_high_peak = []
+                    memory_low_peak = []
+                    memory_list = []
+                    start_time = time.time()
+                    while int(time.time() - start_time) < qga_mem_leak_timeout:
+                        while qga_memory_latest >= qga_memory_previous:
+                            qga_memory_previous = qga_memory_latest
+                            time.sleep(5)
+                            qga_handles_latest, qga_memory_latest = _process_qga_resource()
+                            memory_list.append(qga_memory_latest)
+                            if qga_memory_latest < qga_memory_previous:
+                                memory_high_peak.append(qga_memory_previous)
+                            if int(time.time() - start_time) > qga_mem_leak_timeout:
+                                break
+
+                        while (int(time.time() - start_time) < qga_mem_leak_timeout and
+                               qga_memory_latest <= qga_memory_previous):
+                            qga_memory_previous = qga_memory_latest
+                            time.sleep(5)
+                            qga_handles_latest, qga_memory_latest = _process_qga_resource()
+                            memory_list.append(qga_memory_latest)
+                            if qga_memory_latest > qga_memory_previous:
+                                memory_low_peak.append(qga_memory_previous)
+                    stability_Index, downtrend_value = _compare_peak_value(memory_list, general_list=True)
+                    if downtrend_value == 0:
+                        test.fail("QGA commands caused memory leak completely."
+                                  " memory is %s" % qga_memory)
+                    high_peak_downtrend_value = _compare_peak_value(memory_high_peak)
+                    low_peak_downtrend_value = _compare_peak_value(memory_low_peak)
+                    if high_peak_downtrend_value == 0 and low_peak_downtrend_value == 0:
+                        if stability_Index == 0:
+                            test.fail("QGA commands caused resource leak anyway,"
+                                      " memory is %s" % qga_memory)
+                    # Generally the qga handles won not exceed 300, though
+                    # it run out of 300 sometimes, it would fall back to 300.
+                    if qga_handles_latest > qga_handle_threshold:
+                        bg._stop()
```

Comment:
I doubt if this can work as you expected.
Maybe set the thread as a daemon.

Reply:
@6-dehan Do you still prefer to keep this?

Reply:
nope, has been deleted.